### PR TITLE
fix(artifacts): retire amazon2

### PR DESCRIPTION
**Pull request by yaronkaikov**
Based on scylladb/scylladb#18631 (comment)

Fixes: scylladb/scylla-cluster-tests#8946

Need https://github.com/scylladb/scylla-pkg/pull/4566 to get merged first

### CODE REVIEW

**Comment by fruch** [comment]
@yaronkaikov maybe we should backport to release where it's failing ? i.e. 2024.2 ?

**Review by fruch — CHANGES_REQUESTED** [review]
- `configurations/arm/amazon2.yaml` can be removed
- `sdcm/utils/distro.py` KNOWN_OS can be updated to remove `2` from `'("AMAZON", "amzn", ["2", "2023"], DistroBase.RHEL),`

**Comment by yaronkaikov** [comment]
> * `configurations/arm/amazon2.yaml` can be removed
> * `sdcm/utils/distro.py` KNOWN_OS can be updated to remove `2` from `'("AMAZON", "amzn", ["2", "2023"], DistroBase.RHEL),`

Done

File: sdcm/cluster_aws.py [hunk 1]
```diff
@@ -873,10 +872,6 @@ def traffic_control(self, tcconfig_params=None):
             self.remoter.run('sudo bash -cxe "%s"' % tc_command)
 
     def install_traffic_control(self):
```
Comment (Annamikhlin) [inline sdcm/cluster_aws.py:874]: seems `install_traffic_control` function is not needed any more

Comment (yaronkaikov) [reply sdcm/cluster_aws.py:874]: it is been used by `nemesis.py`

Comment (fruch) [reply sdcm/cluster_aws.py:874]: as yaron was saying, it is being using in nemesis.py.

and isn't related to this PR

**Review by fruch — APPROVED** [review]
LGTM

File: sdcm/cluster_aws.py (hunk 1 above)
Comment (Annamikhlin) [reply sdcm/cluster_aws.py:874]: But now after this change the function is not installing anything... so the the is not exactly correct
and looking as the following:
```
    def install_traffic_control(self):
        return self.remoter.run("/sbin/tc -h", ignore_status=True).ok
```

Just checking and returns ok... 
and it was added only for amazon2...

**Comment by yaronkaikov** [comment]
@roydahan if you decide to add `backport/2024.2` it must also be aligned with pkg, otherwise we will continue triggering this job

**Comment by roydahan** [comment]
So please align, because I deleted these jobs from 2024.2

**Pull request merged**